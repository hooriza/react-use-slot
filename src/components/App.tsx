import MyComponent from './MyComponent';
import { Fragment } from "../module";

const App = () => {
  return (
    <div className="App">
      <MyComponent>
        <h1 className="foo" slot="header">H1 in <code>header</code> slot</h1>
        <h1>H1 in <code>default</code> slot</h1>
        <h2>H2 in <code>default</code> slot</h2>
        <Fragment slot="header">
          Fragment in <code>header</code> slot
        </Fragment>
        {1234}
        hello
        <h2 slot="world">
          H2 in <code>world</code> slot
        </h2>
        world
        <strong>strong in <code>default</code> slot</strong>
        <div>div in <code>default</code> slot</div>
        <li slot="listitem">
          {(item: string) => <Fragment>{item}</Fragment>}
        </li>
      </MyComponent>
    </div>
  );
}

export default App;
