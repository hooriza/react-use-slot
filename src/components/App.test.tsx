import renderer from "react-test-renderer";
import App from "./App";

test("test with sample code", () => {
  expect(renderer.create(<App />)).toMatchSnapshot();
});
