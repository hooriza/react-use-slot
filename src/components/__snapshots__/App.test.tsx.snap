// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with sample code 1`] = `
<div
  className="App"
>
  <div>
    <fieldset>
      <legend>
        header
      </legend>
      <h1
        className="foo"
      >
        H1 in 
        <code>
          header
        </code>
         slot
      </h1>
      Fragment in 
      <code>
        header
      </code>
       slot
    </fieldset>
    <fieldset>
      <legend>
        world
      </legend>
      <h2>
        H2 in 
        <code>
          world
        </code>
         slot
      </h2>
    </fieldset>
    <fieldset>
      <legend>
        default
      </legend>
      <h1>
        H1 in 
        <code>
          default
        </code>
         slot
      </h1>
      <h2>
        H2 in 
        <code>
          default
        </code>
         slot
      </h2>
      1234
      hello
      world
      <strong>
        strong in 
        <code>
          default
        </code>
         slot
      </strong>
      <div>
        div in 
        <code>
          default
        </code>
         slot
      </div>
    </fieldset>
    <fieldset>
      <legend>
        fallback
      </legend>
      fallback
    </fieldset>
    <fieldset>
      <legend>
        loop
      </legend>
      <ul>
        <li>
          foo
        </li>
        <li>
          bar
        </li>
        <li>
          baz
        </li>
      </ul>
    </fieldset>
  </div>
</div>
`;
